app/src/main/java/gateway/controller/CtrlFileList.java

```diff
@@ -0,0 +1,95 @@
+package gateway.controller;
+
+import gateway.config.Config;
+import gateway.services.ServiceAuth;
+import gateway.soap.request.ReqFileList;
+import gateway.soap.response.File;
+import gateway.soap.response.ResFileList;
+import gateway.soap.response.ResStatus;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.UUID;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class CtrlFileList
+{
+	public static ResFileList file_list (ReqFileList args)
+	{
+		// Create a new instance of ResFileList to store the results
+		ResFileList resFileList = new ResFileList ();
+		UUID userUUID;
+		String url;
+
+		// User authentication
+		ResStatus resAuth = ServiceAuth.authenticate (args.token);
+		if (resAuth.error) {
+			return ResStatus.downCast (ResFileList.class, resAuth);
+		}
+
+		// Get the user's UUID from the authentication token
+		userUUID = UUID.fromString (ServiceAuth.tokenGetClaim (args.token, "uuid"));
+
+		// Build the URL for the request.
+		if (args.location == null) {
+			// If no location is specified, build the URL without a parentUUID parameter
+			url = Config.getMetadataBaseUrl () + "/files/list/" + userUUID;
+		} else {
+			// If a location is specified, include the parentUUID parameter
+			url = Config.getMetadataBaseUrl () + "/files/list/" + userUUID +
+				  "?parentUUID=" + args.location;
+		}
+
+		try {
+			// Make an HTTP GET request
+			HttpResponse<String> response = HttpClient.newHttpClient ().send (
```

Comment:
Shouldn't this call and the data parsing be done from the metadata service? @Woynert

Reply:
You're right thinking it would be more organized, but actually we are reserving the ServiceMetadata class to store commonly used endpoints.

For one time uses cases like this one it's fine to do it in the CtrlFileList. This is in order to avoid cluttering the ServiceMetadata class with every single endpoint.

This approach it's fine for this case since these proxy like functions are trivial.